sdk/js/src/api/stream/subscribeToWebSocketStreams.js

```
@@ -0,0 +1,101 @@
+
+  // Interweaving multiple streams has the side effect of making the standard event listener
+  // state machine look erratic externally - only certain streams may fail, while others may continue
+  // to work. Upper layers which use the stream must not be exposed to this detail - once one
+  // stream fails, we need to ensure that all streams are closed and no stray events are emitted.
+  //
+  // The standard state machine guarantees that once an error or close event is emitted, no further
+  // message events will be emitted. It is also guaranteed that once an error event is emitted, a close
+  // event will follow.
+  let [closeAll, hadError, hadClose] = [() => {}, false, false]
+  const uniformListeners = {
+    [EVENTS.MESSAGE]: (...params) => {
+      if (hadClose || hadError) {
+        return
+      }
+      notify(filledListeners[EVENTS.MESSAGE], ...params)
+    },
+    [EVENTS.ERROR]: (...params) => {
+      if (hadClose || hadError) {
+        return
+      }
+      hadError = true
+      notify(filledListeners[EVENTS.ERROR], ...params)
+    },
+    [EVENTS.CLOSE]: (...params) => {
+      if (hadClose) {
+        return
+      }
+      hadClose = true
+      notify(filledListeners[EVENTS.CLOSE], ...params)
+      closeAll()
+    },
+  }
+
+  const pendingStreams = baseUrls.map(baseUrl =>
+    subscribeToWebSocketStream(payload, baseUrl, uniformListeners, endpoint, timeout),
+  )
+  try {
+    const streams = await Promise.all(pendingStreams)
+    closeAll = () => Promise.all(streams.map(stream => stream.close()))
+    return {
+      open: () => streams.forEach(stream => stream.open()),
+      close: closeAll,
+    }
+  } catch (error) {
+    // Ensure that if only some streams fail, the successful ones are closed.
+    await Promise.all(
+      pendingStreams.map(async pendingStream => {
+        try {
+          const stream = await pendingStream
+          await stream.close()
+        } catch {}
```

In the JS codebase, we typically include a comment into the empty catch to explain why this catch is ignored. This is to signal that the pass is voluntarily and not an oversight.